test/e2e/specs/editor/blocks/cover.spec.js

```diff
@@ -226,6 +226,72 @@ test.describe( 'Cover', () => {
 		await expect( overlay ).toHaveCSS( 'background-color', 'rgb(0, 0, 0)' );
 		await expect( overlay ).toHaveCSS( 'opacity', '0.5' );
 	} );
+
+	test( 'z-index of Navigation block inside a Cover block is higher than the other cover inner blocks', async ( {
+		editor,
+		page,
+		coverBlockUtils,
+	} ) => {
+		// Insert a Cover block
+		await editor.insertBlock( { name: 'core/cover' } );
+		const coverBlock = editor.canvas.getByRole( 'document', {
+			name: 'Block: Cover',
+		} );
+
+		// Choose a color swatch to transform the placeholder block into
+		// a functioning block.
+		await coverBlock
+			.getByRole( 'option', {
+				name: 'Color: Black',
+			} )
+			.click();
+
+		// Insert a Navigation block inside the Cover block
+		await editor.selectBlocks( coverBlock );
+		await coverBlock.getByRole( 'button', { name: 'Add block' } ).click();
+		await page.keyboard.type( 'Navigation' );
+		const blockResults = page.getByRole( 'listbox', {
+			name: 'Blocks',
+		} );
+		const blockResultOptions = blockResults.getByRole( 'option' );
+		await blockResultOptions.nth( 0 ).click();
+
+		// Insert a second Cover block.
+		await editor.insertBlock( { name: 'core/cover' } );
+		const secondCoverBlock = editor.canvas
+			.getByRole( 'document', {
+				name: 'Block: Cover',
+			} )
+			.last();
+
+		// Choose a color swatch to transform the placeholder block into
+		// a functioning block.
+		await secondCoverBlock
+			.getByRole( 'option', {
+				name: 'Color: Black',
+			} )
+			.click();
+
+		// Set the viewport to a small screen where the menu will become a modal and open it.
+		await page.setViewportSize( { width: 375, height: 1000 } );
+		const navigationBlock = editor.canvas.getByRole( 'document', {
+			name: 'Block: Navigation',
+		} );
+		await editor.selectBlocks( navigationBlock );
+		await editor.canvas
+			.getByRole( 'button', { name: 'Open menu' } )
+			.click();
+
+		const menuZIndex = await coverBlockUtils.calculateZIndexContext(
+			coverBlock.locator( '.wp-block-navigation__responsive-container' )
```

Comment:
Is there an accessible target we can test against instead of using implementation details like CSS class selectors? `getByRole` is generally preferred when selecting elements.

Reply:
This doesn't have any accessible semantics. We could use a data-test-id that is only added in certain environments?

Reply:
Is this something that could be addressed in a code quality follow-up to this PR?